workers/data_refinery_workers/processors/test_compendia.py

```diff
@@ -325,16 +273,428 @@ def test_create_compendia_danio(self):
+            {
+                "organism": danio_rerio,
+                "accession_code": "GSM1487222",
+                "title": "this sample will be filtered",
+                "technology": "RNA-SEQ",
+                "filename": None,
+            },
+            experiment,
+        )
+        rnas.append(sample.accession_code)
+
+        dset = Dataset()
+        dset.data = {"GSE1234": micros, "GSE5678": rnas}
+        dset.scale_by = "NONE"
+        dset.aggregate_by = "SPECIES"
+        dset.svd_algorithm = "ARPACK"
+        dset.quantile_normalize = True
+        dset.save()
+
+        pjda = ProcessorJobDatasetAssociation()
+        pjda.processor_job = job
+        pjda.dataset = dset
+        pjda.save()
+
+        imputation_index = create_compendia.COMPENDIA_PIPELINE.index(
+            create_compendia._perform_imputation
+        )
+
+        pipeline = Pipeline(name=PipelineEnum.CREATE_COMPENDIA.value)
+        job_context = utils.run_pipeline(
+            {"job_id": job.id, "pipeline": pipeline},
+            create_compendia.COMPENDIA_PIPELINE[:imputation_index],
+        )
+
+        # First, run the imputation step without removing anything to get a baseline
+        expected_context = utils.run_pipeline(
+            job_context.copy(), [create_compendia.COMPENDIA_PIPELINE[imputation_index]]
+        )
+
+        # Now pick some rows to remove according to the instructions from
+        # https://github.com/AlexsLemonade/refinebio/pull/2879#issuecomment-895143336
+
+        random.seed(42)
+
+        # Select some rows randomly and mask ~30% of the values
+        rare_rows = random.sample(list(job_context["microarray_matrix"].index), k=25)
+        rare_genes = {}
+        for row in rare_rows:
+            cols = random.sample(list(job_context["microarray_matrix"].columns), k=110)
+            rare_genes[row] = cols
+            for col in cols:
+                job_context["microarray_matrix"].loc[row, col] = np.nan
+
+        # Now some entries from the other rows to mask completely at random
+        individual_indices = random.sample(
+            list(
+                itertools.product(
+                    set(job_context["microarray_matrix"].index) - set(rare_rows),
+                    job_context["microarray_matrix"].columns,
+                )
+            ),
+            k=1000,
+        )
+        for row, col in individual_indices:
+            job_context["microarray_matrix"].loc[row, col] = np.nan
+
+        final_context = utils.run_pipeline(
+            job_context, [create_compendia.COMPENDIA_PIPELINE[imputation_index]]
+        )
+        self.assertDidNotFail(job)
+
+        index = set(final_context["merged_no_qn"].index) & set(
+            expected_context["merged_no_qn"].index
+        )
+        columns = set(final_context["merged_no_qn"].columns) & set(
+            expected_context["merged_no_qn"].columns
+        )
+
+        N = 0
+        se = 0
```

Comment:
What does `se` mean? Is there a more descriptive name we can use? Maybe `error_squared`?